internal/transport/console.go

```diff
@@ -7,24 +7,22 @@ import (
 	"sync"
 	"time"
 
+	api "github.com/flightctl/flightctl/api/v1alpha1"
 	"github.com/flightctl/flightctl/internal/flterrors"
 	"github.com/flightctl/flightctl/internal/store"
 	"github.com/go-chi/chi/v5"
 	"github.com/gorilla/websocket"
 )
 
-var upgrader = websocket.Upgrader{
-	CheckOrigin: func(r *http.Request) bool {
-		return true // Allow connections from any origin
-	},
-}
-
 func (h *WebsocketHandler) HandleDeviceConsole(w http.ResponseWriter, r *http.Request) {
 	orgId := store.NullOrgId
 	deviceName := chi.URLParam(r, "name")
 
-	h.log.Debugf("websocket console connection requested for device: %s", deviceName)
-	consoleSession, err := h.consoleSessionManager.StartSession(r.Context(), orgId, deviceName)
+	h.log.Infof("websocket console connection requested for device: %s", deviceName)
+
+	// Extract metadata
+	metadata := r.URL.Query().Get(api.DeviceQueryConsoleSessionMetadata)
```

Comment:
for the `ws` connection, i'd rather use the ws native subprotocols header, then encoding subprotocol into url query. This header is also supported when opening ws connection from browsers.

Reply:
The negotiation is between the agent and the cli.  The cli uses ws but the agent doesn't.  The the protocols are used by the agent to select a specific protocol.  This then is returned by gRPC, and used by the server in the standard way to indicate the selected protocol.

Reply:
OK